core/src/snapshot.rs

```diff
@@ -92,79 +88,68 @@ impl SnapshotMaker {
+                iroha_logger::info!(at_height, "Successfully created a snapshot of WSV");
                 self.new_wsv_available = false;
             }
             Ok(Err(error)) => {
-                iroha_logger::error!(%error, "Failed to create snapshot for wsv.");
+                iroha_logger::error!(%error, "Failed to create a snapshot of WSV");
             }
             Err(panic) => {
-                iroha_logger::error!(%panic, "Task panicked during creation of wsv snapshot.");
+                iroha_logger::error!(%panic, "Task panicked during creation of WSV snapshot");
             }
         }
     }
 
-    /// Serialize and write snapshot to file,
-    /// overwriting any previously stored data.
+    /// Create from [`Config`].
     ///
-    /// # Errors
-    /// - IO errors
-    /// - Serialization errors
-    fn try_write_snapshot(wsv: &WorldStateView, snapshot_dir: impl AsRef<Path>) -> Result<()> {
-        let path_to_file = snapshot_dir.as_ref().join(SNAPSHOT_FILE_NAME);
-        let path_to_tmp_file = snapshot_dir.as_ref().join(SNAPSHOT_TMP_FILE_NAME);
-        let file = std::fs::OpenOptions::new()
-            .create(true)
-            .write(true)
-            .truncate(true)
-            .open(&path_to_tmp_file)
-            .map_err(|err| Error::IO(err, path_to_tmp_file.clone()))?;
-        let mut serializer = serde_json::Serializer::new(file);
-        wsv.serialize(&mut serializer)?;
-        std::fs::rename(path_to_tmp_file, &path_to_file)
-            .map_err(|err| Error::IO(err, path_to_file.clone()))?;
-        Ok(())
-    }
-
-    /// Create [`Self`] from [`Configuration`]
-    pub fn from_config(config: &Config, sumeragi: SumeragiHandle) -> Self {
-        Self {
-            sumeragi,
-            snapshot_create_every: config.create_every,
-            snapshot_dir: config.store_dir.clone(),
-            snapshot_creation_enabled: config.creation_enabled,
-            new_wsv_available: false,
+    /// Might return [`None`] if the configuration is not suitable for _making_ snapshots.
+    pub fn from_config(config: &Config, sumeragi: &SumeragiHandle) -> Option<Self> {
+        if let Mode::Normal = config.mode {
+            Some(Self {
+                sumeragi: sumeragi.clone(),
+                create_every: config.create_every,
+                store_dir: config.store_dir.clone(),
+                new_wsv_available: false,
+            })
+        } else {
+            None
         }
     }
 }
 
-/// Try deserialize [`WorldStateView`] from snapshot file
+/// Try to deserialize [`WorldStateView`] from a snapshot file.
 ///
 /// # Errors
 /// - IO errors
 /// - Deserialization errors
 pub fn try_read_snapshot(
-    snapshot_dir: impl AsRef<Path>,
+    store_dir: impl AsRef<Path>,
     kura: &Arc<Kura>,
     query_handle: LiveQueryStoreHandle,
     BlockCount(block_count): BlockCount,
-) -> Result<WorldStateView> {
+) -> Result<TryReadResult> {
```

Comment:
```suggestion
) -> Result<WorldStateView, TryReadError> {
```

I think this is more in line with Rust conventions. Just group all possible errors in one enum

Reply:
The thing is, it isn't really an error. 

Anyway, would it be fine?

```rust
enum TryReadError {
  NotFound,
  Other(Error), // general `snapshot::Error`
}
```

Reply:
> The thing is, it isn't really an error.

you want to say that `NonFound`is not necessarily an error?

> Anyway, would it be fine?

yes

Reply:
> The thing is, it isn't really an error.

it is an error. Because it's returned from `try_read_snapshot`. An alternative would be to check for existence on disk before calling `try_read_snapshot`

Reply:
Done. Also split `Error` into `TryReadError` and `TryWriteError`. They have an overlap, but the difference is bigger.